### ISSUE TITLE
Implement a user-friendly build/deploy system

### ISSUE THREAD

**Issue by belen-albeza** [mozdevs/boo#8]
Options:

a) Forget about the `dist` folder, move the inside of `app` to the root directory and commit building artifacts such as `bundle.js` to the repo. Then the `gh-pages` branch is identical to `master` and we don't need to mess up with git subtrees. I don't think this is such a terrible idea since the only artifact we have right now is `bundle.js`.

b) Keep the `dist` folder and try to tweak the scripts to find a friendly way to push into `gh-pages` a subtree with `dist`. I tried this today and it when there were conflicts they had to be solved manually (i.e. "complicated")

c) Keep the current, messy approach just for ourselves and offer an alternative way of deploying. I think that we can assume that if people have npm in their systems they will also have rsync. I don't know about Windows users, though.

d) Keep the current messy approach just for ourselves and don't offer any way of deploying into an external server :sunglasses:  (we would keep the building command that generates the `dist` folder, though).

Thoughts?

**Comment by sole** [comment]
- a) **No**, because: We don't want to spread "bad practices". Checking in `bundle.js` along with the source files is not a very good idea unless you're a library (although it lets you try the project without having to `npm install`)
- b) **verging towards yes** What "conflicts", exactly? I'm pretty sure it can be done with just an npm command (`npm run deploy`)--Bespoke.js used `grunt-gh-pages` which can do this. It says [it creates a temporary clone and etc](https://github.com/tschaub/grunt-gh-pages#overview).
- c) **yes** but obviously keep the build process as described in d)
- d) **no**, we need to let people know how this works. This is a learning material and it's supposed to be used, not to be left to die in obscurity because no one knows how to build and deploy it :-)

In short: `npm run deploy` should do the deploy for you automatically. If it spits out errors then you can follow whatever manual instructions we provide to fix them, but it should work for 99% of the cases. This way anyone can learn how automatic `gh-pages` deploys are done.

We should also add to the README that "deploying" is copying the `dist/` folder to a web server with an SFTP client, rsync or any other file copying mechanism.

Related, once it's done: we should probably turn this into a sample/template project (e.g. has source code, builds a bundle.js, deploys to `gh-pages`)

**Comment by sole** [comment]
I also found https://www.npmjs.com/package/gh-pages

**Comment by sole** [comment]
I've noticed that the current build system is outputting `bundle.js` to the same `app/js` folder, this is very confusing and also doesn't match the instructions for `dist`.

I have [an example of how to do a whole build](https://github.com/sole/reveal-minimal/blob/master/package.json) making sure directories exist and cleaning them before etc.

**Comment by sole** [comment]
Filed https://github.com/mozdevs/boo/issues/11 that needs to be done before we continue with this :)
